### PR TITLE
Bug/12308

### PR DESCRIPTION
**Pull request by moraj-turing**
This PR implements missing requirements from #12308  

# Impact

- [ ] New migrations
- [x] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs
N/A

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

**Comment by kadewu** [comment]
Hi @moraj-turing, thank you for your contribution!

About changes, please adjust PR name to something more meaningful (e.g. the issue name).

Is this PR ready for review? Cause from my understanding it's still missing some of the requirements from the ticket. You can always mark PR as draft when you still working on it, and this way it will be more clear for other developers.
Please also remember to test changes that you are introducing.

**Comment by moraj-turing** [comment]
Hello, this PR is ready. Most of the work was already done on previous changes by other contributors.

File: saleor/graphql/product/mutations/product/product_media_create.py [hunk 1]
```diff
@@ -31,13 +35,31 @@ class ProductMediaCreateInput(BaseInputObjectType):
...
+    )
+
     class Meta:
         doc_category = DOC_CATEGORY_PRODUCTS
 
 
 class ProductMediaCreate(BaseMutation):
     product = graphene.Field(Product)
     media = graphene.Field(ProductMedia)
+    support_meta_field = True
```
Comment (kadewu) [inline saleor/graphql/product/mutations/product/product_media_create.py:61]: Can you please add tests to check ensure that changes works?

File: saleor/graphql/product/types/products.py [hunk 2]
```diff
@@ -939,6 +939,7 @@ class Product(ChannelContextTypeWithMetadata[models.Product]):
     )
     variants = NonNullList(
         ProductVariant,
+        required=True,
```
Comment (kadewu) [inline saleor/graphql/product/types/products.py:942]: Please do not mix different issues together in one PR

**Review by kadewu — COMMENTED** [review]
And couple more general comments:
- please install `pre-commit` locally to run checks. More can be read here: https://docs.saleor.io/docs/3.x/developer/community/contributing#coding-style
- above will run check that graphql schema changes are not included, to generate schema you can ran `npm run build-schema`
- please update `CHANGELOG.md` file 
- I don't see changes regarding firing the webhook, will that be included?

**Comment by aniav** [comment]
Hey @moraj-turing there hasn't been any activity on this PR for some time. I'll go ahead and close it. Feel free to reopen when you address the comments above. Thank you!

**Pull request closed without merging**